### PR TITLE
[new DL][Navigation] Apply background treatment to hover/focus/active states, make disabled state look more… disabled

### PR DESCRIPTION
**Pull request by kaelig**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-react/issues/3385
Fixes https://github.com/Shopify/polaris-react/issues/3432
Fixes https://github.com/Shopify/polaris-react/issues/3434

### WHAT is this pull request doing?

- Hover/focused/active states were declared and overridden various times, this PR simplifies code to remove most unnecessary overrides
- The `Navigation` component has many possible states, of which very few were documented. This adds an example to show most things possible with that component at a glance, also making spotting regressions easier.
- Fixes states to match designs from https://github.com/Shopify/polaris-react/issues/3385 and https://github.com/Shopify/polaris-react/issues/3432

| [Before](https://5d559397bae39100201eedc1-uvqcbwoenl.chromatic.com/?path=/story/all-components-navigation--navigation-with-various-states-and-secondary-elements) | [After](https://5d559397bae39100201eedc1-atnoracqnk.chromatic.com/?path=/story/all-components-navigation--navigation-with-various-states-and-secondary-elements) |
| ------ | ----- |
| Screen Shot 2020-10-14 at 10 30 51 PM | Screen Shot 2020-10-14 at 10 31 11 PM |
| Screen Shot 2020-10-14 at 10 31 54 PM | Screen Shot 2020-10-14 at 10 31 56 PM |

High contrast mode in Edge/Win 10:

Screen Shot 2020-10-14 at 10 37 30 PM

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

#### Compare these stories

- [Before](https://5d559397bae39100201eedc1-uvqcbwoenl.chromatic.com/?path=/story/all-components-navigation--navigation-with-various-states-and-secondary-elements)
- [After](https://5d559397bae39100201eedc1-atnoracqnk.chromatic.com/?path=/story/all-components-navigation--navigation-with-various-states-and-secondary-elements)

#### Todo

- Check all selected/hover/focus/active states with mouse **and keyboard** make sense and match the new design.
- Check everything works as expected on iOS (on mobile I could only test on Android)
- Check high contrast mode in various browsers

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by kaelig** [comment]
@jessebc these disabled styles are achieved by reducing opacity of the elements, a technique also used in the old nav.

What do you think?

(there may be accessibility concerns with the resulting color contrast)

File: src/components/Navigation/Navigation.scss [hunk 1]
```diff
@@ -229,92 +217,41 @@ $disabled-fade: 0.6;
...
-        var(--p-action-primary, color('indigo')),
-        var(--p-override-transparent, color('white')),
-        filter('action')
-      );
+      @media (-ms-high-contrast: active) {
+        @include recolor-icon(
+          $fill-color: color('white'),
+          $filter-color: filter('white')
+        );
+      }
```
Comment (kaelig) [inline src/components/Navigation/Navigation.scss:239]: Small tweak to make icons easier to spot in Edge/Win10 on hover. Sadly Firefox turns them into black and they become really hard to spot on hover in that browser.

**Review by alex-page — APPROVED** [review]
This looks great @kaelig, thanks for adding the additional example to the README. Was there a reason we couldn't use `--p-text-disabled`?

**Comment by jessebc** [comment]
> @jessebc these disabled styles are achieved by reducing opacity of the elements, a technique also used in the old nav.
> 
> What do you think?
> 
> (there may be accessibility concerns with the resulting color contrast)

Did you try using the `disabled` variants of text and icon?

**Review by kyledurand — APPROVED** [review]
This looks great, my only slight worry is using opacity for disabled states. Not a blocker though, we can iterate on that

**Comment by alex-page** [comment]
@kaelig I am going to merge this in and cut a release candidate. If you want to keep iterating we will cut another release.

**Comment by kaelig** [comment]
Thank you for the fast review! ❤️ 

**Re: disabled text** – As it turns out, `--p-text-disabled` was already being applied to disabled items, which doesn't contrast as much with surrounding non-disabled text.

Screen Shot 2020-10-15 at 7 09 30 AM

**Re: opacity** – the opacity technique is used in purpler and works quite well visually. If we prefer to use tokens without opacity here, it may warrant some tweaks at the color system level.

**Re: icons** – I haven't tried applying `--p-icon-disabled`, because we'd need to add a filter for it to polaris-tokens (we use manually generated CSS filters to recolor icons coming from the GraphQL API, such as the "sales channels" ones). In the interest of time, I went for a double-opacity trick – I suggest exploring this part in a future iteration.

Let me know if you have any other questions!

**Pull request merged**